mlir/lib/Dialect/MIOpen/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -2636,18 +2636,29 @@ struct GridwiseGemmV2RewritePattern
                                       enableOutSwizzles);
     }
 
-    op.erase();
+    b.eraseOp(op);
 
     return success();
   }
 };
 
 void LowerMIOpenOpsStep2Pass::runOnOperation() {
   MLIRContext *ctx = &getContext();
+  ConversionTarget target(*ctx);
+  target.addIllegalOp<miopen::GridwiseGemmOp>();
+  target.addIllegalOp<miopen::GridwiseGemmV2Op>();
+
+  target.addLegalDialect<arith::ArithmeticDialect>();
+  target.addLegalDialect<miopen::MIOpenDialect>();
+  target.addLegalDialect<AffineDialect>();
+
   RewritePatternSet patterns(ctx);
-  patterns.add<GridwiseGemmRewritePattern, GridwiseGemmV2RewritePattern>(ctx);
-  if (failed(applyPatternsAndFoldGreedily(getOperation(), std::move(patterns))))
+  patterns.add<GridwiseGemmRewritePattern>(ctx);
+  patterns.add<GridwiseGemmV2RewritePattern>(ctx);
```

Comment:
Why not keep the old `add` call? Also, does `addIllegalOp` take multiple arguments?

Reply:
Yes for both. I don't see much of the difference from callee perspective. Looking at the `RewritePatternSet` source code I feel like they are more or less equivalent with each other. Or is there any compelling reason to adopt the variadic version?

Reply:
Because that's the style preferred upstream

Reply:
Can't argue with that :-p, addressed in 2d45f48702c046ebe4ffd5aed75ca288afda9157